src/components/AppHeader.tsx

```diff
@@ -35,15 +36,15 @@ interface AppHeaderCallbackProps {
 const mapStateToProps = (state: XCSpyState): AppHeaderProps => {
     return {
         currentComponent: state.components.currentComponent,
-        getStateMachines: (component) => {
-            let initialized = state.components.initialized;
+        getStateMachines: (component: string): string[] => {
+            const initialized = state.components.initialized;
             if (!initialized)
                 return [];
-            let componentProperties = state.components.componentProperties;
+            const componentProperties = state.components.componentProperties;
             return Object.keys(componentProperties[component].stateMachineProperties);
         },
-        getComponents: () => {
-            let initialized = state.components.initialized;
+        getComponents: (): string[] => {
```

Comment:
I still don't understand why you need a function. A property called components should be fine